src/config/config.go

```diff
@@ -42,3 +44,11 @@ func Parse(config []byte) (cfg Config, err error) {
 	}
 	return
 }
+
+func (cfg Config) WriteFile(path string, mode os.FileMode) (err error) {
```

Comment:
The provider doesn't return an opaque blob, it returns a Config.

I was treating the cache as an opaque blob by having the config package be responsible for the parsing/marshaling of the struct, that's exactly what you objected to.

Having the engine perform the marshal diminishes the opacity by dictating it's json, while relying on config.Parse() to continue to accept the json.

I like the way I have it now, with the cached format being private to the engine on both the marshal/unmarshal sides, totally isolated from the implementation in config.